### DISCUSSION TITLE
blog/hawkeye-esbuild-analyzer

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#22] [General]
# blog/hawkeye-esbuild-analyzer

Effortlessly analyze your JS bundles and uncover actionable insights to boost performance and enhance user experience.

https://angularexperts.io/blog/hawkeye-esbuild-analyzer

**Comment by markgoho** [comment]
the web app link leads nowhere

**Reply by nivekcode** [reply]
Thx @markgoho the link should be fixed now

**Comment by amosISA** [comment]
Hi, nice tool but the browser is not working. When I upload stats.json is not working. In the console appears some error regarding: "Cannot read properties of undefined (reading 'length')". It's a build from an app from an NX monorepo.

**Reply by nivekcode** [reply]
Okay, I have to look into that, can you share some more details on your setup? Its NX but are you using Angular, React or something else? Do you use Webpack or EsBuild? Note that this is an esbuild analyzer so webpack's `stats.json` files do not work.

**Reply by amosISA** [reply]
Yes it's a webpack build. I'll try with esbuild.

**Comment by radekbusa** [comment]
Hello Nivek, we've got an intranet enterprise app, approx. 500 kLoC with a considerably big monolithic shared module (hundreds of components). Hawkeye is definitely very pleasant with respect to filtering bundle sizes and lazy/eager module differentiation, but when I try opening the shared module in graph view, the module graph becomes illegible (lots of very narrow stripes representing individual source files and I believe it overflows outside of the graphical view to the right).

**Reply by nivekcode** [reply]
Thanks for the interesting input! I’ll definitely look into this. I assume your app is internal, so sharing the stats.json might not be feasible, right? If it were open source, though, that would be super helpful for reproduction. Of course, only if it’s possible and aligns with your project’s policies. 😊
